src/main/java/org/whispersystems/textsecuregcm/sqs/ContactDiscoveryQueueSender.java

```diff
@@ -0,0 +1,89 @@
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.AmazonSQSClientBuilder;
+import com.amazonaws.services.sqs.model.SendMessageRequest;
+import com.amazonaws.services.sqs.model.MessageAttributeValue;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.SharedMetricRegistries;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.whispersystems.textsecuregcm.configuration.ContactDiscoveryConfiguration;
+import org.whispersystems.textsecuregcm.util.Constants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+public class ContactDiscoveryQueueSender {
+
+  private static final Logger logger = LoggerFactory.getLogger(ContactDiscoveryQueueSender.class);
+
+  private final MetricRegistry metricRegistry    = SharedMetricRegistries.getOrCreate(Constants.METRICS_NAME);
+  private final Meter          serviceErrorMeter = metricRegistry.meter(name(ContactDiscoveryQueueSender.class, "serviceError"));
+  private final Meter          clientErrorMeter  = metricRegistry.meter(name(ContactDiscoveryQueueSender.class, "clientError"));
+
+  private final String queueUrl;
+  private final AmazonSQS sqs;
+
+  public ContactDiscoveryQueueSender(ContactDiscoveryConfiguration config) {
+    final AWSCredentials credentials = new BasicAWSCredentials(config.getAccessKey(), config.getAccessSecret());
+    final AWSStaticCredentialsProvider credentialsProvider = new AWSStaticCredentialsProvider(credentials);
+    this.queueUrl = config.getQueueUrl();
+    this.sqs = AmazonSQSClientBuilder.standard().withCredentials(credentialsProvider).build();
+  }
+
+  private void sendMessage(String action, String user) {
+    final Map<String, MessageAttributeValue> messageAttributes = new HashMap<>();
+    messageAttributes.put("id", new MessageAttributeValue().withDataType("String").withStringValue(user));
+    messageAttributes.put("action", new MessageAttributeValue().withDataType("String").withStringValue(action));
+    SendMessageRequest send_msg_request = new SendMessageRequest()
+            .withQueueUrl(queueUrl)
+            .withMessageBody("-")
+            .withMessageDeduplicationId(user + action)
+            .withMessageGroupId(user)
+            .withMessageAttributes(messageAttributes);
+    try {
+      sqs.sendMessage(send_msg_request);
+    } catch (AmazonServiceException ex) {
+      serviceErrorMeter.mark();
+      logger.warn("sqs service error: ", ex);
+    } catch (AmazonClientException ex) {
+      clientErrorMeter.mark();
+      logger.warn("sqs client error: ", ex);
+    }
+  }
+
+
+  public void addRegisteredUser(String user) {
+    sendMessage("add", user);
+  }
+
+  public void deleteRegisteredUser(String user) {
+    sendMessage("delete", user);
+  }
```

Comment:
Public methods above private methods please

Reply:
done in a10eeecd4478c83a7e14029b498d4f9e3dae1b6b